### PR TITLE
[front] fix(csv) - csv parsing

### PR DESCRIPTION
**Pull request by aubin-tchoi**
## Description

- Follow up on logs gathered using https://github.com/dust-tt/dust/pull/9118.
- `values[h]` is a function: [https://app.datadoghq.eu/logs?query=%40dd.env%3Aprod%20%22Error%20parsing%20record%22[…]=stream&from_ts=1733318910000&to_ts=1733318920000&live=false](https://app.datadoghq.eu/logs?query=%40dd.env%3Aprod%20%22Error%20parsing%20record%22&agg_m=count&agg_m_source=base&agg_t=count&cols=host%2Cservice&event=AwAAAZOR26ROwDl6NwAAABhBWk9SMjY1dUFBQUJKMkxSdXl1VWhnQVMAAAAkMDE5MzkxZGItY2Y1Ni00YjA5LTliYmMtZDQyYzQzZmVjOWU5AAAjhQ&fromUser=true&messageDisplay=inline&refresh_mode=paused&storage=hot&stream_sort=desc&viz=stream&from_ts=1733318910000&to_ts=1733318920000&live=false)
- Hypothesis here is that header names are stuff like `onString`.
- Also handles falsey values correctly (does not replace them with "" systematically, only deals with `null` and `undefined`).
- More logs [here](https://app.datadoghq.eu/logs?query=%22Error%20uploading%20table%20to%20Dust.%22&agg_m=count&agg_m_source=base&agg_t=count&cols=host%2Cservice&fromUser=true&messageDisplay=inline&refresh_mode=paused&storage=hot&stream_sort=desc&viz=stream&from_ts=1733319867925&to_ts=1733319925679&live=false).

`> {}["constructor"]
[Function: Object]`

Tested with:

constructor | toString | 0 | undefined | null | [] | {}
-- | -- | -- | -- | -- | -- | --
0 | 1 | undefined | 2 |   | null |  

## Risk

Low.

## Deploy Plan

- Deploy front.

### CODE REVIEW

File: front/lib/api/tables.ts [hunk 1]
```diff
@@ -369,14 +369,14 @@ export async function rowsFromCsv({
...
-  const valuesByCol: Record<string, string[]> = {};
+  const valuesByCol: Record<string, string[]> = Object.create(null);
   for await (const anyRecord of parser) {
     if (i++ >= rowIndex) {
       const record = anyRecord as string[];
       for (const [i, h] of header.entries()) {
         try {
           valuesByCol[h] ??= [];
-          (valuesByCol[h] as string[]).push(record[i] || "");
+          (valuesByCol[h] as string[]).push(record[i] ?? "");
```
Comment (spolu) [inline front/lib/api/tables.ts:379]: What types of value are we going to get that we didn't get? Are they desirable?

File: front/lib/api/tables.ts [hunk 2]
```diff
@@ -369,14 +369,14 @@ export async function rowsFromCsv({
 
   let i = 0;
   const parser = parse(csv, { delimiter });
-  const valuesByCol: Record<string, string[]> = {};
+  const valuesByCol: Record<string, string[]> = Object.create(null);
```
Comment (spolu) [inline front/lib/api/tables.ts:372]: Can you explain a bit more? :)

File: front/lib/api/tables.ts (hunk 1 above)
Comment (aubin-tchoi) [reply front/lib/api/tables.ts:379]: this change is preemptive and not related to the fix, here if we get a value of 0 in `record[i]` it gets converted to ""

File: front/lib/api/tables.ts (hunk 2 above)
Comment (aubin-tchoi) [reply front/lib/api/tables.ts:372]: mb completely forgot

Comment (aubin-tchoi) [reply front/lib/api/tables.ts:373]: here it was `{}["constructor"]` that caused the issue

Comment (aubin-tchoi) [reply front/lib/api/tables.ts:373]: https://app.datadoghq.eu/logs?query=%22Error%20uploading%20table%20to%20Dust.%22&agg_m=count&agg_m_source=base&agg_t=count&cols=host%2Cservice&fromUser=true&messageDisplay=inline&refresh_mode=paused&storage=hot&stream_sort=desc&viz=stream&from_ts=1733319867925&to_ts=1733319925679&live=false

Comment (spolu) [reply front/lib/api/tables.ts:373]: :+1:

**Review by spolu — APPROVED** [review]

**Pull request merged**